cosmic-usage-ui/src/main/resources/public/app/main.js

```diff
@@ -36,11 +49,73 @@ const app = {
 
         $(this.monthSelectorComponent).datepicker('setDate', new Date());
         this.renderPrintingHeaders();
+        this.renderDomainTableHeaders();
         this.renderDomainsList();
 
         $(this.generateReportButton).on('click', this.generateReportButtonOnClick);
     },
 
+    sortDomainTableByHeader: function(e) {
+        var currState = $(e.target).attr(this.domainTableHeaderState);
+
+        this.clearAllSortingSigns();
+
+        this.flipSortingSign($(e.target), currState);
+
+        var currentSorting = this.getCurrentSorting();
+
+        this.generateSortableReport(currentSorting[0], currentSorting[1]);
+    },
+
+    clearAllSortingSigns: function() {
+        var that = this;
+        $( this.tableHeaderSortable ).each(function() {
+            that.flipSortingSign($(this), 'DESC');
```

Comment:
Please create constants out of strings like this (`DESC`, `ASC`, `&#9660`, `&#9650`).